janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -6165,4 +6165,234 @@ private void testWritingAndReading(File f) {
+        assertEquals(0,countOfEdges("vtName", "A", "OUT", "egLabel", "assocKind", 2,false));
+        assertEquals(1,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 1, false));
+        assertEquals(0,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 2, false));
+        assertEquals(0,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 1,false));
+        assertEquals(1,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 2,false));
+
+        //Reindexing
+        mgmt.updateIndex(relationIndex, SchemaAction.REINDEX).get();
+        finishSchema();
+
+        relationIndex = mgmt.getRelationIndex(t,indexName);
+        assertEquals(SchemaStatus.ENABLED, relationIndex.getIndexStatus());
+
+        //asserting after reindex
+        assertEquals(0,countOfEdges("vtName", "A", "IN", "egLabel", "assocKind", 1, false));
+        assertEquals(1,countOfEdges("vtName", "A", "IN", "egLabel", "assocKind", 2, false));
+        assertEquals(1,countOfEdges("vtName", "A", "OUT", "egLabel", "assocKind", 1,false));
+        assertEquals(0,countOfEdges("vtName", "A", "OUT", "egLabel", "assocKind", 2,false));
+        assertEquals(1,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 1, false));
+        assertEquals(0,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 2, false));
+        assertEquals(0,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 1,false));
+        assertEquals(1,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 2,false));
+    }
+	
+    private void initReindexSchema() {
+        if(!mgmt.containsEdgeLabel("egLabel")){
+            mgmt.makeEdgeLabel("egLabel").multiplicity(Multiplicity.MULTI).make();
+        }
+
+        if(!mgmt.containsPropertyKey("vtName")) {
+            mgmt.makePropertyKey("vtName").dataType(String.class).cardinality(Cardinality.SINGLE).make();
+        }
+		
+        if(!mgmt.containsPropertyKey("assocKind")) {
+            mgmt.makePropertyKey("assocKind").dataType(Integer.class).cardinality(Cardinality.SINGLE).make();	
+        }
+    }
+    
+    //To make sure no self links created
+    public void assertEdgeCounts(boolean isReindexed) {
+        assertEquals(1,countOfEdges("vtName", "A", "IN", "egLabel", "assocKind", 1, isReindexed));
+        assertEquals(2,countOfEdges("vtName", "A", "OUT", "egLabel", "assocKind", 1,isReindexed));
+        assertEquals(2,countOfEdges("vtName", "B", "IN", "egLabel", "assocKind", 1, isReindexed));
+        assertEquals(0,countOfEdges("vtName", "B", "OUT", "egLabel", "assocKind", 1, isReindexed));
+        assertEquals(1,countOfEdges("vtName", "C", "IN", "egLabel", "assocKind", 1, isReindexed));
+        assertEquals(2,countOfEdges("vtName", "C", "OUT", "egLabel", "assocKind", 1, isReindexed));
+    }
+    
+    public int countOfEdges(String vPropName, String vPropValue, String direction, String edgeLabel, String edgeProperty, int edgeValue, boolean isReindexed) {
```

Comment:
`isReindexed` parameter is not used. Anyway, you don't even need a whole method here. I suggest doing
```java
g.V().has("vtName", "A").inE().hasLabel("egLabel").has("assoKind", 1).count().next()
```
directly, wherever you currently use `countOfEdges` method.

Reply:
testcase removed as this seems to be unnecessary for this PR